File: src/main/java/com/chengte99/Hello.java
package com.chengte99;

public class Hello {
    public static void main(String[] args) {
//        System.out.println("Hello World");
//        new Person().hello();
        String name = "abcde";
        name = null;
        System.out.println(name.charAt(3));
        System.out.println(name.length());
        System.out.println(name.substring(2));
        System.out.println(name.substring(1,4));

        Person p = new Person("Kevin", 70.5f, 1.77f);
//        p.weight = 70.5f;
//        p.height = 1.77f;
        System.out.println(p.bmi());
        p.hello();

        int score = 88;
        System.out.println(score >= 80 && score < 85);
        char c = 'A';
        System.out.println(c > 'a');

        /*int age = 19;
        Integer age2 = 23;
        char c = 'A';
        Character c2 = 'B';
        float weight = 66.5f;
        Float height = 123.3f;
        byte b = 123;
        boolean isEnable = true;
        Boolean isOK = isEnable;
        String name = "kevin";*/
    }
}

File: src/main/kotlin/com/chengte99/Hello.kt
package com.chengte99

fun main() {
//    println("Hello kotlin")
//    Person().hello()
    var name: String? = "abcde";
    name = null
    println(name?.length);
    println(name?.get(3));
    println(name?.substring(3));
    println(name?.substring(2,4));

    val h = Human(70.5f, 1.77f)
    h.hello()
    println(h.bmi())

    val score = 80
    println(score <= 80)
    val c: Char = 'A'
    println(c.toInt() > 60)

    /*var age = 12
    age = 23
    var weight: Float = 55.6f
    var name: String
    name = "Kevin"*/
}

class Human (var weight: Float, var height: Float, var name: String = "") {
    init {
        println("TEST $weight")
    }

//    constructor(name: String, weight: Float, height: Float): this(weight, height)

    fun bmi(): Float{
        var bmi = weight / (height * height)
        return bmi
    }

    fun hello() {
        println("Hello kotlin")
    }
}
